sql/set_var.cc

```diff
@@ -1186,8 +1182,27 @@ int fill_sysvars(THD *thd, TABLE_LIST *tables, COND *cond)
     fields[6]->store(type->str, type->length, scs);
 
     // VARIABLE_COMMENT
-    fields[7]->store(var->option.comment, strlen(var->option.comment),
-                           scs);
+    if (var->option.comment != NULL)
+      fields[7]->store(var->option.comment, strlen(var->option.comment),
+                       scs);
+    else if (var->option.deprecation_substitute != NULL)
+    {
+      if (strcmp(var->option.deprecation_substitute, OPT_DEPRECATED_NO_REPLACEMENT) == 0)
+      {
+        char deprecation_msg[]= "DEPRECATED";
+        fields[7]->store(deprecation_msg, strlen(deprecation_msg), scs);
+      }
+      else
+      {
+        char deprecation_msg[]= "DEPRECATED and replaced by: ";
+        size_t buflen= strlen(deprecation_msg) + strlen(var->option.deprecation_substitute);
+        char *buf= (char *)malloc(buflen);
+
+        strxmov(buf, deprecation_msg, var->option.deprecation_substitute, NullS);
+        fields[7]->store(buf, buflen, scs);
+        free(buf);
+      }
+    }
```

Comment:
I'm not sure about it. `--help` appends lots of various stuff to the comment. Like "Enabled by default use --skip to disable" or "One of: option1, option2, option3", etc. And now "(disabled, use %s instead)". None of the other such --help auto-additions are appended to the comment in INFORMATION_SCHEMA.SYSTEM_VARIABLES.

I don't think the disabled comment should be an exception. Better add a separate column to the I_S table.